tests/unit_tests/c_event.cc

```diff
@@ -195,10 +195,25 @@ TEST_F(CEventTest, DISABLED_EventWaitNotTriggeredEvent)
   realm_runtime_t runtime = *runtime_impl;
   ASSERT_REALM(realm_user_event_create(runtime, &event));
 
-  realm_status_t status = realm_event_wait(runtime, event, nullptr);
+  realm_status_t status = realm_event_wait(runtime, event, INT64_MAX, nullptr);
   EXPECT_EQ(status, REALM_SUCCESS);
 }
 
+TEST_F(CEventTest, DISABLED_EventWaitTimeoutEvent)
```

Comment:
We might need to have a test case that triggering the event within the max_ns, however, TBH, I have no idea how to implement it within the gtest. Since the wait will block the calling thread, we will need to call the trigger in another thread